catalogs/catalog-fileset-hadoop/src/main/java/com/datastrato/gravitino/catalog/file/HadoopCatalogOperations.java

```diff
@@ -0,0 +1,383 @@
+          .withName(ident.name())
+          .withComment(schemaEntity.comment())
+          .withProperties(schemaEntity.properties())
+          .withAuditInfo(schemaEntity.auditInfo())
+          .build();
+
+    } catch (NoSuchEntityException exception) {
+      throw new NoSuchSchemaException("Schema " + ident.name() + " does not exist", exception);
+    } catch (IOException ioe) {
+      throw new RuntimeException("Failed to load schema " + ident.name(), ioe);
+    }
+  }
+
+  @Override
+  public Schema alterSchema(NameIdentifier ident, SchemaChange... changes)
+      throws NoSuchSchemaException {
+    try {
+      if (!store.exists(ident, Entity.EntityType.SCHEMA)) {
+        throw new NoSuchSchemaException("Schema " + ident.name() + " does not exist");
+      }
+    } catch (IOException ioe) {
+      throw new RuntimeException("Failed to check if schema " + ident.name() + " exists", ioe);
+    }
+
+    try {
+      SchemaEntity entity =
+          store.update(
+              ident,
+              SchemaEntity.class,
+              Entity.EntityType.SCHEMA,
+              schemaEntity -> updateSchemaEntity(ident, schemaEntity, changes));
+
+      return new HadoopSchema.Builder()
+          .withName(ident.name())
+          .withComment(entity.comment())
+          .withProperties(entity.properties())
+          .withAuditInfo(entity.auditInfo())
+          .build();
+
+    } catch (IOException ioe) {
+      throw new RuntimeException("Failed to update schema " + ident.name(), ioe);
+    } catch (NoSuchEntityException nsee) {
+      throw new NoSuchSchemaException("Schema " + ident.name() + " does not exist", nsee);
+    } catch (AlreadyExistsException aee) {
+      throw new RuntimeException("Schema " + ident.name() + " already exists", aee);
+    }
+  }
+
+  @Override
+  public boolean dropSchema(NameIdentifier ident, boolean cascade) throws NonEmptySchemaException {
+    try {
+      SchemaEntity schemaEntity = store.get(ident, Entity.EntityType.SCHEMA, SchemaEntity.class);
+      Map<String, String> properties =
+          Optional.ofNullable(schemaEntity.properties()).orElse(Collections.emptyMap());
+
+      String schemaLocation =
+          (String)
+              SCHEMA_PROPERTIES_METADATA.getOrDefault(
+                  properties, HadoopSchemaPropertiesMetadata.LOCATION);
+      Path schemaPath =
+          Optional.ofNullable(schemaLocation)
+              .map(Path::new)
+              .orElse(catalogStorageLocation.map(p -> new Path(p, ident.name())).orElse(null));
+
+      // Nothing to delete if the schema path is not set.
+      if (schemaPath == null) {
+        return true;
+      }
+
+      FileSystem fs = schemaPath.getFileSystem(hadoopConf);
+      // Nothing to delete if the schema path does not exist.
+      if (!fs.exists(schemaPath)) {
+        return true;
+      }
+
+      if (fs.listStatus(schemaPath).length > 0) {
+        if (!cascade) {
+          throw new NonEmptySchemaException(
```

Comment:
Can we output at least one sub-entity for troubleshooting purposes?

Reply:
IMO, I think It's confusing to only output one, either we output all or we output nothing.